src/test/scala/com/softwaremill/realworld/users/UserEndpointTestSupport.scala

```diff
@@ -0,0 +1,152 @@
+package com.softwaremill.realworld.users
+
+import com.softwaremill.diffx.{Diff, compare}
+import com.softwaremill.realworld.users
+import com.softwaremill.realworld.utils.TestUtils.backendStub
+import sttp.client3.testing.SttpBackendStub
+import sttp.client3.ziojson.*
+import sttp.client3.{HttpError, Response, ResponseException, basicRequest}
+import sttp.model.Uri
+import sttp.tapir.server.stub.TapirStubInterpreter
+import zio.test.Assertion.*
+import zio.test.{Assertion, TestResult, assertTrue, assertZIO}
+import zio.{RIO, Random, ZIO, ZLayer}
+
+object UserEndpointTestSupport {
+
+  def callGetCurrentUser(
+      authorizationHeader: Map[String, String],
+      uri: Uri
+  ): ZIO[UsersEndpoints, Throwable, Either[ResponseException[String, String], User]] =
+    ZIO
+      .service[UsersEndpoints]
+      .map(_.getCurrentUser)
+      .flatMap { endpoint =>
+        basicRequest
+          .get(uri)
+          .headers(authorizationHeader)
+          .response(asJson[User])
```

Comment:
especially for tests maybe it would be easier if in case of a response error (connection error /parsing exception), we returned a failed effect, instead of an `Either`? This can be done using `.getRight`https://sttp.softwaremill.com/en/stable/responses/exceptions.html?highlight=getRight#deserialization-errors

Reply:
I like it ;) I'll will add it to the issues